packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx

```diff
@@ -64,6 +65,24 @@ export class DetailsHeaderBase extends BaseComponent<IDetailsHeaderBaseProps, ID
     sourceIndex: number;
     targetIndex: number;
   };
+
+  public static getDerivedStateFromProps(newProps: IDetailsHeaderBaseProps, prevState: IDetailsHeaderState): IDetailsHeaderState {
+    const columnReorderProps: IColumnReorderHeaderProps | undefined =
+      newProps.columnReorderProps || (newProps.columnReorderOptions && getLegacyColumnReorderProps(newProps.columnReorderOptions));
+
+    const newState: IDetailsHeaderState = { columnReorderProps };
+
+    if (newProps.isAllCollapsed !== undefined) {
+      newState.isAllCollapsed = newProps.isAllCollapsed;
+    }
+
+    if (newProps.groupNestingDepth !== prevState.groupNestingDepth) {
+      newState.groupNestingDepth = newProps.groupNestingDepth;

```

Comment:
why not just always return it? is this condition needed?

Reply:
Consolidated that logic since I'm in agreement with you. Looking at the previous logic and the fact that `setState` calls are batched, it could have been consolidated then also.